### PR TITLE
[feature]add fmtlib(#2291)

### PR DESCRIPTION
**Pull request by Tangruilin**
### What problem does this PR solve?

Issue Number: #2291  

Problem Summary:

### What is changed and how it works?

What's Changed:

add libfmt fot glog

How it Works:
now we can use `fmt::print("{}", 4)` to format log. Details can look issue#2291

Side effects(Breaking backward compatibility? Performance regression?):

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Comment by Tangruilin** [comment]
recheck

**Comment by Tangruilin** [comment]
are there someone review this PR

**Comment by fansehep** [comment]
Can you use ```fmt``` in a cpp file? let the build deps have the fmt.

BTW. give a example to how to print a ```class``` by libfmt in a cpp file?

**Comment by wuhongsong** [comment]
@Cyber-SiKu

File: WORKSPACE [hunk 1]
```diff
@@ -37,6 +37,25 @@ git_repository(
     commit = "d12de388c97998f5ccd5cb97ed0da728815ef438",
 )
 
+git_repository(
+    name = "fmt",
+    branch = "master",
```
Comment (Cyber-SiKu) [inline WORKSPACE:42]: Use a stable version? For example tag: 9.1.0.

**Review by Cyber-SiKu — CHANGES_REQUESTED** [review]

**Comment by Cyber-SiKu** [comment]
> Can you use `fmt` in a cpp file? let the build deps have the fmt.
> 
> BTW. give a example to how to print a `class` by libfmt in a cpp file?



agree! You can modify an old code as an example, or modify some unit tests.

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Tangruilin** [comment]
> Can you use `fmt` in a cpp file? let the build deps have the fmt.
> 
> BTW. give a example to how to print a `class` by libfmt in a cpp file?

ok, i will do it these day

**Comment by Cyber-SiKu** [comment]
@Tangruilin  Will you continue to follow up this PR?

**Comment by ilixiaocui** [comment]
@Tangruilin Will you continue to follow up this PR?

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
how to use:https://github.com/fmtlib/fmt/blob/master/README.rst

**Review by Cyber-SiKu — APPROVED** [review]

File: WORKSPACE [hunk 2]
```diff
@@ -37,6 +37,25 @@ git_repository(
...
+    remote = "https://github.com/fmtlib/fmt",
+    commit = "bce8d4ed087a0560492426f9f5be2713804daded"
```
Comment (Cyber-SiKu) [inline WORKSPACE:44]: miss `,`

**Comment by Cyber-SiKu** [comment]
continue in [2635](https://github.com/opencurve/curve/pull/2635)

**Pull request closed without merging**